### PR TITLE
BUGFIX: ConvertUrisImplementation throws an error when trying to convert uris when requesting the site in a format different from html

### PR DESCRIPTION
**Pull request by Benjamin-K**
**What I did**
I changed the LinkingService to always create node uris in html format. This is required because otherwise requests with a format different from html will most likely throw an exception.

**How I did it**
I changed the requested format in the function call of "createNodeUri" in the TYPO3.Neos-LinkingService. An other option would be to make the required format customizable in the `TYPO3.Neos:ConvertUris` EEL-Helper.

**How to verify it**
- Create a custom format (in my case: json.search).
- Try to create a http uri for a node using the `TYPO3.Neos:ConvertUris` EEL-Helper:
  
  ```
  link = ${'node://' + someNode.identifier}
  link.@process.convertUris = TYPO3.Neos:ConvertUris
  ```

This will throw the following exception:

```
Could not resolve a route and its corresponding URI for the given parameters. 
This may be due to referring to a not existing package / controller / action 
while building a link or URI. Refer to log and check the backtrace for more details.

root<TYPO3.TypoScript:Case>/
  …
    link/
      __meta/
        process/
          convertUris<TYPO3.Neos:ConvertUris>/
```

**Checklist**
- [x] Code follows the PSR-2 coding style
- [X] Tests have been created, run and adjusted as needed
- [X] The PR is created against the 2.1-Branch [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)

### CODE REVIEW

**Comment by Benjamin-K** [comment]
The added Test needs some fixes in L320-325 as it does not use the requested format correctly. So there should be some changes made.

File: TYPO3.Neos/Tests/Unit/TypoScript/ConvertUrisImplementationTest.php [hunk 1]
```diff
@@ -301,4 +301,31 @@ public function evaluateReplaceResourceLinkTargets()
...
+        $this->mockWorkspace->expects($this->any())->method('getName')->will($this->returnValue('live'));
+        $this->mockActionRequest->expects($this->any())->method('getFormat')->will($this->returnValue('json.something'));
+
+        $self = $this;
+        $this->mockLinkingService->expects($this->atLeastOnce())->method('resolveNodeUri')->will($this->returnCallback(function ($nodeUri) use ($self, $nodeIdentifier) {
+            if ($nodeUri !== 'node://' . $nodeIdentifier) {
+                $self->fail('Unexpected node URI "' . $nodeUri . '"');
+            }
+            return 'http://www.example.org/some-node-link';
+        }));
```
Comment (aertmann) [inline TYPO3.Neos/Tests/Unit/TypoScript/ConvertUrisImplementationTest.php]: wait how is this testing that `createNodeUri` is called with 'html' when it's mocking the `resolveNodeUri` call? as far as I understand this test, it doesn't actually test the changed behavior..

does it fail if you remove the actual change?

**Comment by aertmann** [comment]
change makes sense, but the test seem a bit off

**Comment by jonnitto** [comment]
@Benjamin-K what is the status here?

File: TYPO3.Neos/Tests/Unit/TypoScript/ConvertUrisImplementationTest.php (hunk 1 above)
Comment (Benjamin-K) [reply TYPO3.Neos/Tests/Unit/TypoScript/ConvertUrisImplementationTest.php]: Sorry for not replying to this for a long time. I need to recheck the changes i made then. Is the used format (always html) the prefered way or should there be an option to customize the format used?
However, i need some help with the tests when it comes to an update.

**Pull request closed without merging**